### PR TITLE
[77] Add support for instrumented descriptors

### LINKED ISSUE #77: object_as_dict doesn't work with ORM-instrumented descriptor queries
[`object_as_dict`](https://github.com/nestauk/daps_utils/blob/dab9fa3a1b7a621a6e7559632982004f5444f3fa/daps_utils/db.py#L16-L19) is handy for taking a record from a query and converting it to a python dict. But it can only do this on queries formed by passing the Base ORM, e.g. `session.query(RawJobAd)`. If you instead query specific columns with `session.query(RawJobAd.id, RawJobAd.description)`, you get a non-Inspectable return instance that causes an exception.

I added a version that can handle both cases to my ojd_daps branch [here](https://github.com/nestauk/ojd_daps/blob/4b909011edd8444c233d63dfca6f44d79ca9eb17/ojd_daps/flows/enrich/labs/requires_degree/model/io.py#L49-L62).

If this looks promising I can do a little PR to bring it back to daps_utils.

### PR DESCRIPTION
**Pull request by jooh**
More flexible object_as_dict for queries that select specific columns.

Closes #77 

Would benefit from a test.

### CODE REVIEW

**Review by jaklinger — APPROVED** [review]
tldr: good to merge

It would indeed benefit from a test- and this is on the to-do list of #29, but deemed ok at the time as it was a near direct port from the old DAPS, plus the ORM and db stuff is actually quite hard to test without comprehensive spin-up/tear-down fixtures... which is why I've assigned it as a task in it's own right. In short, we can forgive a lack of tests at this stage until we've set that up!
